### PR TITLE
[14.0][REF] account_payment_return_import_iso20022: Adapt module to native payment refactoring

### PR DESCRIPTION
**Pull request by pedrobaeza**
Following https://github.com/OCA/bank-payment/pull/979

Blocked by:

- [x] https://github.com/OCA/bank-payment/pull/979

@Tecnativa

### CODE REVIEW

File: account_payment_return_import_iso20022/models/payment_return.py [hunk 1]
```diff
@@ -9,10 +10,20 @@ class PaymentReturnLine(models.Model):
...
-        lines = self.filtered(lambda x: not x.move_line_ids and x.reference)
-        for line in lines:
-            line.move_line_ids = AccountMoveLine.search(
-                [("bank_payment_line_id.name", "=", line.reference)]
+        """Include in the matches the lines coming from payment orders."""
+        matched = self.env["payment.return.line"]
+        for line in self.filtered(lambda x: not x.move_line_ids and x.reference):
+            payments = self.env["account.payment"].search(
+                [
+                    ("payment_reference", "=", line.reference),
```
Comment (carlosdauden) [inline account_payment_return_import_iso20022/models/payment_return.py]: payment_reference is not sended as end2end_identification:
https://github.com/OCA/bank-payment/pull/979/files#diff-514ea70d8407133fa56949b0111cc1ff8befe4aea3d5160c6863f2099a8927aaR159

EndToEndId is the one used by the iso20022 scheme to indicate which payment is returned and for this reason it is  established as a reference when importing the file:
https://github.com/OCA/account-payment/blob/c1f620c97127d2fa6191f6464165685355a5e17c/account_payment_return_import_iso20022/wizard/pain_parser.py#L62-L64

The implemented search may be fine as an alternative search, but the main search must be based on what is sent with the EndToEndId tag.

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: account_payment_return_import_iso20022/models/payment_return.py (hunk 1 above)
Comment (pedrobaeza) [reply account_payment_return_import_iso20022/models/payment_return.py]: OK, I didn't know about the usage of `EndToEndId`. Then it's going to be even easier. I'll change it.

Comment (pedrobaeza) [reply account_payment_return_import_iso20022/models/payment_return.py]: Fixed, and added compatibility with old approach (both in `account_payment_order` and here).

**Comment by fredaspen1** [comment]
> Following https://github.com/OCA/bank-payment/pull/979
> 
> Blocked by:
> 
> - [x] https://github.com/OCA/bank-payment/pull/979
> 
> @Tecnativa 

Hi Pedro, we are looking for dev, to build a payment platform for banks using crypto as a way of payment ans comply with the iso20022 standard

**Pull request merged**